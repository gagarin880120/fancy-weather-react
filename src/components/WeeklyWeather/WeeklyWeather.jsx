import React from 'react';
import styles from './WeeklyWeather.module.css';

export default function WeeklyWeather() {
  return (
    <div className={styles.wrapper}>
      {/* получая из хранилища информацию рендерить массив из карточек */}
      <WeeklyWewatherCard />
    </div>
  );
}
