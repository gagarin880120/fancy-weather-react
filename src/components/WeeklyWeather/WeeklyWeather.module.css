.wrapper {
  display: flex;
  margin-top: 70px;
  flex-wrap: wrap;
}
