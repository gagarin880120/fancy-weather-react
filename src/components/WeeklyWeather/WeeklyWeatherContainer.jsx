import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import WeeklyWeather from './WeeklyWeather';
import { selectWeeklyWeather } from '../../redux/selectors';

export function WeeklyWeatherContainer({ weeklyWeather }) {
  console.log(weeklyWeather);
  return (
    weeklyWeather
      ? (
        <WeeklyWeather
          weeklyWeather={weeklyWeather}
        />
      ) : <div>Loading...</div>
  );
}

export const mapStateToProps = (state) => ({
  weeklyWeather: selectWeeklyWeather(state),
});

WeeklyWeatherContainer.propTypes = {
  weeklyWeather: PropTypes.instanceOf(Array),
};

WeeklyWeatherContainer.defaultProps = {
  weeklyWeather: [],
};

export default connect(mapStateToProps)(WeeklyWeatherContainer);
