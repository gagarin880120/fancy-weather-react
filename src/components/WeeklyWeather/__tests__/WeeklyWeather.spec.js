import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import TestRenderer, { act } from 'react-test-renderer';
import WeeklyWeather from '../WeeklyWeather';

describe('WeeklyWeather', () => {
  describe('component', () => {
    test('should render without crashing', () => {
      const renderer = new ShallowRenderer();
      renderer.render(
      <WeeklyWeather
        weeklyWeather={[
          {
            datetime: '21.07.2020',
            temp: 15,
            weekday: 'Tuesday',
            icon: 'Sunny'
          },
          {
            datetime: '22.07.2020',
            temp: 18,
            weekday: 'Wednesday',
            icon: 'Clouds'
          },
        ]}
      />
      );

      const result = renderer.getRenderOutput();

      expect(result).toMatchSnapshot();
    });
  });
});
