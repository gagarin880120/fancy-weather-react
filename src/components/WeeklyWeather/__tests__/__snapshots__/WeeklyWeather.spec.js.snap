// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeeklyWeather component should render without crashing 1`] = `
<div
  className="wrapper"
>
  <WeeklyWeatherCard
    icon="Sunny"
    temp={15}
    weekday="Tuesday"
  />
  <WeeklyWeatherCard
    icon="Clouds"
    temp={18}
    weekday="Wednesday"
  />
</div>
`;
